### ISSUE TITLE
Receiving Correct Response but No Authentication

### ISSUE THREAD

**Issue by masonry47** [flipboxfactory/saml-sp#163]
Hey @dsmrt!

We are having a new issue.  In short, all seems to be working correctly but the user does not actually get logged in. We are using a SAML Response tool and see everything come back as we would expect it to.  The login url successfully gets us to the IDP, we are able to successfully authenticate there and then are returned to our site but are not logged in.  We have turned on dev mode to see if there is anything relevant in the logs and there doesn't seem to be.  Really the only thing unique in this setup compared to another one we have done is the IDP is handling the relay URL. And our relay parameter is being ignored.  Could that have anything to do with the issue. 

Thanks as always for you amazing support!

**Comment by dsmrt** [comment]
:wave: @masonry47,

Thanks for the kind words. SAML has it's own unique issues and can be confusing so I try and make up for it with quick responses. 

So, if I understand this correctly, it looks like everything is working but the user is not logged in once the auth process is finished? 

1. Can you see saml logs going through the login process once the IdP returns the user to the Craft site? Once the user's returned to the site (sent back to `/sso/login/<uid>`), logs should show some activity here. Any errors?
2. Via logging or in the SAML Response tool, look at the host / domain / url and make sure they are pointing the same place. For example, if the return url is "https://www.example.com/sso/login" but the user ultimately lands on "https://example.com/sso/login", the cookies might not transfer over correctly, and the user won't be logged in. Watch out for "https" vs "http" here too.

Check these and let me know how this goes.

**Comment by masonry47** [comment]
Thanks!

on #1, assuming. you mean web.log, though a little tough for me to read what is happening in there, there is no errors for sure.  Should we see some of the user attributes in the log? Because we do not.

On #2 are you saying this URL

<img width="461" alt="Screen Shot 2022-03-18 at 11 48 20 AM" src="https://user-images.githubusercontent.com/48130527/159047302-cd7e8c8c-773c-4051-9151-b319da9a675b.png">

Should match where the user is redirected to exaclty?

**Comment by dsmrt** [comment]
1) Here's a good way to tail and grep the contents of the web.log, as events are coming through.

```bash
tail -f storage/logs/web.log | grep -A 10 '\[saml-'
```

2) That "Destination" should match the Assertion Consumer Service (ACS) location so there seems to be a configuration issue there (which is likely the problem). The IdP should be sending the user back to a url similar to `http://example.com/sso/login/<uid>`. There is probably a URL in the IdP configuration that needs to be updated. Hopefully they call this Assertion Consumer Service, but it depends on the IdP. What IdP are you using? Also, see my screenshot below  to know where to find the ACS in your Craft install.

![Screen Shot 2022-03-18 at 11 00 39 AM](https://user-images.githubusercontent.com/1995478/159049441-d81a6b63-70a8-42d5-919d-de82ec02df50.png)

**Comment by masonry47** [comment]
Thanks!  When running the Tail command we just get this on the login event (just replaced urls -- idp.url.com is the IDP and https://sub.site.com is the craft site ).  Meaning after clicking "Login Via IDP" being taken to the IDP login, logging in and then returning to the Craft site (but failing to to login to Craft).  So this doesn't seem like it is telling us much.  The IDP is an instance of Drupal with a plugin installed.  We successfully connected with a staging version of it, so thinking it must be a config issue (see your second point) on the IDP side??.

`2022-03-18 10:29:21 [-][-][-][info][saml-core] Looping thru 1 role descriptors
2022-03-18 10:29:21 [-][-][-][info][saml-core] Using Service method: getSingleSignOnService
2022-03-18 10:29:21 [-][-][-][info][saml-core] Looping thru 1 role descriptors
2022-03-18 10:29:21 [-][-][-][info][saml-core] Using Service method: getSingleSignOnService
2022-03-18 10:29:21 [-][-][-][info][yii\db\Command::query] SELECT *
FROM `craft_sites`
WHERE (`craft_sites`.`dateDeleted` IS NULL) AND (`id`=1)
2022-03-18 10:29:21 [-][-][-][profile begin][yii\db\Command::query] SELECT *
FROM `craft_sites`
WHERE (`craft_sites`.`dateDeleted` IS NULL) AND (`id`=1)
2022-03-18 10:29:21 [-][-][-][profile end][yii\db\Command::query] SELECT *
FROM `craft_sites`
WHERE (`craft_sites`.`dateDeleted` IS NULL) AND (`id`=1)
2022-03-18 10:29:21 [-][-][-][info][yii\db\Command::query] SHOW FULL COLUMNS FROM `craft_sites`
--
2022-03-18 10:29:21 [-][-][-][info][saml-core] Looping thru 1 role descriptors
2022-03-18 10:29:21 [-][-][-][info][saml-core] Using Service method: getSingleSignOnService
2022-03-18 10:29:21 [-][-][-][info][saml-core] Looping thru 1 role descriptors
2022-03-18 10:29:21 [-][-][-][info][saml-core] Using Service method: getSingleSignOnService
2022-03-18 10:29:21 [-][-][-][trace][saml-core] <?xml version="1.0"?>

<samlp:AuthnRequest xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" ID="_01000e998d28e14a8aa9ea1389999fd0f81c50ad23" Version="2.0" IssueInstant="2022-03-18T17:26:10Z" Destination="https://idp.url.com/simplesaml/saml2/idp/SSOService.php" AssertionConsumerServiceURL="https://sub.site.com/sso/login/a9109e83-723b-4619-9b4b-a890aadf9396" ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"><saml:Issuer>https://sub.site.com</saml:Issuer></samlp:AuthnRequest>

2022-03-18 10:26:10 [-][-][-][trace][saml-core] Redirect to 1055 byte URL: https://idp.url.com/simplesaml/saml2/idp/SSOService.php?SAMLRequest=hZJRT8IwEMe%2FytL3re2GsDZAgvIgCUbC0AdfTLcd0GRra68j%2Bu0diBF9wD5c0t79rv%2F758ao2sbJWRf2Zg1vHWCI3tvGoDwlJqTzRlqFGqVRLaAMlSxmD0uZJkw6b4OtbEMukOuEQgQftDUkWswn5JVxxhgIkddpDnygcqUEKJ7loj%2Fbmm1zXt0wVacZiZ7BY09OSN%2BoxxE7WBgMyoT%2BiaVpzLKY5xs%2BkulQcvZConk%2FjTYqnKh9CA4lpc76oJqkbDpwXmObVLalqFvXwFE%2FPYaU6trRongswB90BYnbOxLNvsXfWYNdC%2F6cfVovf9p3Rh%2BOQsPH3y%2FQ0sbutKFKcCYgz%2BJRmpXxYMhFLMpBGatcMKXqrcjEkESrs7m32tTa7K77Wn4VobzfbFbxGmrtoQpkOj5OI09e%2Bel%2FEsf0svrr9ns1pp8%3D&RelayState=aHR0cHM6Ly91bml2ZXJzaXR5LmJsdWVwcmlzbS5jb20vYWRtaW4v&SigAlg=http%3A%2F%2Fwww.w3.org%2F2001%2F04%2Fxmldsig-more%23rsa-sha256&Signature=L48GJFXI%2F9eXs04a1nkYSIRPFVBi2aEM6ZbwT%2B5uDsUEIgu4i54Ecf54vNBa6foBe1denbgXcLFoJKmWFo8m2gnr6R7TFT6RD702CLJ6LzlfXsOlLy%2Bar4oeu7oyQ6XWS8edRMYbkvy%2FYyp6Gn8SUA2q76KC1OwdGfXebMKhyAgHAu3fOBigJVfAVm5%2Fm1wVizE7xaV3ZfDwEbS3RvMcukZ5Oyns%2FrTctB4cuuDLvFgd8BuofNakmXZkgwV5ApiBaDHrLOHywcODCatCZgEv0H6qOGzYDUgzU%2BHvMPc9ru3tAsH4GcW80xcx6on56Z5vPjNEdd6ZLv3mpYLSfbR%2B7g%3D%3D
2022-03-18 10:26:10 [-][-][-][info][application] $_GET = [
    'p' => 'sso/login/request/69cac80d-6798-46f1-a34d-b8d8734b2f12'
    'RelayState' => 'https://sub.site.com/admin/'
]
$_POST = []

$_FILES = []

$_COOKIE = [
tail: storage/logs/web.log: file truncated`

**Comment by dsmrt** [comment]
That’s just the authnreqest which is builds the request and sends it through the browser to the IdP.   I don’t see logging for the Response which is inline with no. 2 of my previous comment. 

> That "Destination" should match the Assertion Consumer Service (ACS) location so there seems to be a configuration issue there (which is likely the problem). The IdP should be sending the user back to a url similar to http://example.com/sso/login/<uid>. There is probably a URL in the IdP configuration that needs to be updated. Hopefully they call this Assertion Consumer Service, but it depends on the IdP. What IdP are you using? Also, see my screenshot below to know where to find the ACS in your Craft install.

The IdP is incorrectly configured to return the use to the home page and not the /sso/login endpoint (which processes the saml stuff). Fix the ACS url on the IdP side.

**Comment by masonry47** [comment]
got it.  makes sense!  We will check on that config and thanks as always.
